frappe/www/printview.py

```diff
@@ -162,7 +162,13 @@ def get_rendered_template(
 		def get_template_from_string():
 			return jenv.from_string(get_print_format(doc.doctype, print_format))
 
-		if print_format.custom_format:
+		template = None
+		if hook_func := frappe.get_hooks("get_print_format_template"):
+			template = frappe.get_attr(hook_func[-1])(jenv=jenv, print_format=print_format)
+
+		if bool(template):
```

Comment:
Should be fine without the explicit casting as well

Reply:
i can remove it 😅 not really required just added because value is returned from custom apps.